Plugins/GitUIPluginInterfaces/IProcess.cs

```diff
@@ -55,10 +56,17 @@ public interface IProcess : IDisposable
         /// <returns>A task that yields the process's exit code, or <c>null</c> if this object was disposed before the process exited.</returns>
         Task<int> WaitForExitAsync();
 
+        /// <summary>
+        /// Instructs the process component to wait for the associated process to exit, or for the cancellationToken to be cancelled.
+        /// </summary>
+        /// <param name="token">An optional token to cancel the asynchronous operation.</param>
+        /// <returns>A task that will complete when the process has exited, cancellation has been requested, or an error occurs.</returns>
+        Task WaitForExitAsync(CancellationToken token);
```

Comment:
Is cancellation expected to kill the process, or just stop the wait? It's not clear from the docs here.

Reply:
Doc copied from the library doc. I can clarify.
Cancellation throws, the statement after this never executes.

Reply:
I believe it is killed - I cannot see the process running after cancellation and `_process.Exited` is invoked